### PR TITLE
fix(event-sources): handle dynamodb null type as none, not bool

### PR DESCRIPTION
**Pull request by michaelbrewer**
**Issue #, if available:**
- #928

## Description of changes:

Return for `None` for null types:

Example: 

```json5
{
  "Records": [
    {
      ...
      "dynamodb": {
        "OldImage": {
          ...
          "Message": {
            "S": "New item!"
          },
          "Id": {
            "N": "101"
          }
        },
        "Keys": {
          "Id": {
            "N": "101"
          }
        },
        "NewImage": {
          ...
          "Message": {
            "NULL": true
          },
          "Id": {
            "N": "101"
          }
        },
      }
    }
  ]
}
```

Below `null_value` and `get_value` would have returned `True`, and now they return None.

```python3
from aws_lambda_powertools.utilities.data_classes import event_source
from aws_lambda_powertools.utilities.data_classes.dynamo_db_stream_event import (
    DynamoDBStreamEvent,
)

@event_source(data_class=DynamoDBStreamEvent)
def handler(event: DynamoDBStreamEvent, context):
    record = next(event.records)
    message_value = record.dynamodb.new_image.get("message")

    # get_value would be the preferred way to get the underlying value regardless of type
    assert message_value.get_value is None

    # Otherwise when you want to use `null_value` you should first change against `get_type`
    if message_value.get_type == AttributeValueType.Null:
        assert message_value.null_value is None
```

**Checklist**

 

* [X] [Meet tenets criteria](https://awslabs.github.io/aws-lambda-powertools-python/#tenets)
* [X] Update tests
* [X] Update docs
* [X] PR title follows [conventional commit semantics](https://github.com/awslabs/aws-lambda-powertools-python/blob/376ec0a2ac0d2a40e0af5717bef42ff84ca0d1b9/.github/semantic.yml#L2)

## Breaking change checklist

Before `null_value` was returning a boolean,  `True` when `NULL` value was set to `true` and `False` when `NULL` value was not set (or was `False`)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/functional/test_data_classes.py [hunk 1]
```diff
@@ -544,9 +544,19 @@ def test_dynamo_attribute_value_null_value():
     attribute_value = AttributeValue(example_attribute_value)
 
     assert attribute_value.get_type == AttributeValueType.Null
+    assert attribute_value.null_value is None
     assert attribute_value.null_value == attribute_value.get_value
 
 
+def test_dynamo_attribute_value_null_value_invalid():
+    example_attribute_value = {"NULL": False}
```
Comment (heitorlessa) [inline tests/functional/test_data_classes.py:552]: shouldn't this be `True`?

https://github.com/boto/boto3/blob/7a10e21c452e0cb0a6e3d245f7389f1500d2abcf/tests/unit/dynamodb/test_types.py#L69

Comment (michaelbrewer) [reply tests/functional/test_data_classes.py:552]: @michaelbrewer - We are deserializing,  `{"NULL": False}` should never happen (so should we raise a `ValueError` or just ignore that?)

Comment (michaelbrewer) [reply tests/functional/test_data_classes.py:552]: @heitorlessa - i can remove that test, i just wanted it to be clear that we don't valid 

```
    @property
    def null_value(self) -> None:
        """An attribute of type Null.

        Example:
            >>> {"NULL": True}
        """
        return None
```

Comment (michaelbrewer) [reply tests/functional/test_data_classes.py:552]: which is the same as:

```
    def _deserialize_null(self, value):
        return None
```

**Comment by michaelbrewer** [comment]
> Question on whether `NULL` will ever be `False`, boto tests always caters for `True`
> 
> Type table only considers `True` so perhaps DynamoDB doesn't compute that: [boto/boto3@`cf62d26`/boto3/dynamodb/types.py#L85](https://github.com/boto/boto3/blob/cf62d26cb0f28357c935d541ed3f0dae196cd26d/boto3/dynamodb/types.py#L85)

Because boto does not allow you to call `_deserialize_null` directly but we allow for it via `null_value`

**Review by heitorlessa — APPROVED** [review]

**Pull request merged**